files/setup.sh

```diff
@@ -30,6 +30,10 @@ else
     GATEWAY_PROPERTY=$(vmtoolsd --cmd "info-get guestinfo.ovfEnv" | grep "guestinfo.gateway")
     DNS_SERVER_PROPERTY=$(vmtoolsd --cmd "info-get guestinfo.ovfEnv" | grep "guestinfo.dns")
     DNS_DOMAIN_PROPERTY=$(vmtoolsd --cmd "info-get guestinfo.ovfEnv" | grep "guestinfo.domain")
+    PROXY_ENABLED_PROPERTY=$(vmtoolsd --cmd "info-get guestinfo.ovfEnv" | grep "guestinfo.enableproxy")
```

Comment:
This is actually not needed because if the user is explicitly filling out the proxy settings, they're telling us they want this configured. We just need to do some pre-checks and then fill in the right areas based on the input.